message/messaging_service.cc

```diff
@@ -491,7 +491,16 @@ future<> messaging_service::stop_client() {
 
 future<> messaging_service::shutdown() {
     _shutting_down = true;
-    return when_all(stop_nontls_server(), stop_tls_server(), stop_client()).discard_result();
+    auto verbose_stop_action = [] (sstring entity_name, std::function<future<>()> stop_func) -> future<> {
+        mlogger.debug("Stopping {}", entity_name);
+        return stop_func().then([entity_name] {
+            mlogger.debug("Stopping {} Done", entity_name);
+        });
+    };
+
+    return when_all(verbose_stop_action("nontls server", std::bind(&messaging_service::stop_nontls_server, this))
+            , verbose_stop_action("tls server", std::bind(&messaging_service::stop_tls_server, this))
+            , verbose_stop_action("client", std::bind(&messaging_service::stop_client, this))).discard_result();
```

Comment:
Why not simply add the logging into the corresponding functions instead of creating this wrapper?

Reply:
Avoiding code duplication. But I am not very opinionated.
But for my own education: Why is it better?

Reply:
What code duplication you avoid with that? Typing `verbose_stop_action`/`std::bind` three times is not the same code duplication as typing `then`?

It is better because it is readable.

Reply:
OK

Reply:
Fixed